src/public/modules/forms/admin/controllers/view-responses.client.controller.js

```diff
@@ -153,20 +170,97 @@ function ViewResponsesController(
         })
       })
       .catch(function (error) {
-        try {
-          const { errorCount, errorMessage } = JSON.parse(error.message)
-          Toastr.error(
-            errorMessage ||
-              `Error downloading. ${errorCount} response(s) could not be decrypted. Please try again later.`,
-            { timeOut: 3000 },
+        const isCircuitTripped = CircuitBreaker.isOurError(error)
+        if (!isCircuitTripped) {
+          try {
+            const { errorCount, errorMessage } = JSON.parse(error.message)
+            Toastr.error(
+              errorMessage ||
+                `Error downloading. ${errorCount} response(s) could not be decrypted. Please try again later.`,
+              { timeOut: 3000 },
+            )
+          } catch (error) {
+            Toastr.error(`Error downloading. Please try again later.`, {
+              // Should still show error message to let user know download has failed
+              timeOut: 3000,
+            })
+            console.error(
+              'Unknown download encrypted responses error:\t',
+              error,
+            )
+          }
+        } else {
+          // TODO: During soft launch, downloads will proceed even in breaker open state
+          // When circuit breaker is hard launched, replace this code chunk with toastr informing user to try again in XX minutes
+          // Use else here instead of fallback because fallback will trigger once circuit trips, even on the first failed call
+          vm.awaitDownload = true
+          let params = {
+            formId: vm.myform._id,
+            formTitle: vm.myform.title,
+          }
+
+          GTag.downloadWhenBreakerOpen(params) // Tag GA if download attempt is made when breaker is open
+
+          if (vm.datePicker.date.startDate && vm.datePicker.date.endDate) {
+            params.startDate = moment(
+              new Date(vm.datePicker.date.startDate),
+            ).format('YYYY-MM-DD')
+            params.endDate = moment(
+              new Date(vm.datePicker.date.endDate),
+            ).format('YYYY-MM-DD')
+          }
+
+          vm.csvDownloading = true
+          Submissions.downloadEncryptedResponses(
+            params,
+            vm.encryptionKey.secretKey,
           )
-        } catch (error) {
-          console.error('Unknown download encrypted responses error:\t', error)
+            .then(function (result) {
+              $timeout(function () {
+                const { expectedCount, successCount, errorCount } = result
```

Comment:
this should be decomposed into a helper function instead of copying and pasting the code

Reply:
edited